unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarsLODController.cs

```diff
@@ -0,0 +1,147 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL
+{
+    public class AvatarsLODController : MonoBehaviour
+    {
+        public static AvatarsLODController i { get; private set; }
+
+        private const float LODS_LOCAL_Y_POS = 1.8f;
+        private const float LODS_VERTICAL_MOVEMENT = 0.1f;
+        private const float LODS_VERTICAL_MOVEMENT_DELAY = 1f;
+
+        public float lodDistance = 16f;
+        public int maxNonLODAvatars = 20;
+
+        private List<AvatarShape> avatarsList = new List<AvatarShape>();
+        private float lastLODsVerticalMovementTime = -1;
+
+        void Awake()
+        {
+            if (i != null)
+            {
+                Destroy(gameObject);
+                return;
+            }
+            i = this;
+
+            enabled = false;
+            KernelConfig.i.EnsureConfigInitialized()
+                        .Then(config =>
+                        {
+                            // We need this check because in the CI the KernelConfig tests may have this object destroyed somehow
+                            if (i == null || this == null)
+                                return;
+
+                            if (config.features.enableAvatarLODs)
+                            {
+                                enabled = true;
+                                CommonScriptableObjects.playerUnityPosition.OnChange += OnMainPlayerReposition;
+                            }
+                            else
+                            {
+                                enabled = false;
+                            }
+                        });
+        }
+
+        public void LateUpdate()
+        {
+            int listCount = avatarsList.Count;
+
+            bool applyVerticalMovement = Time.timeSinceLevelLoad - lastLODsVerticalMovementTime > LODS_VERTICAL_MOVEMENT_DELAY;
+            GameObject lodGO;
+            for (int i = 0; i < listCount; i++)
+            {
+                lodGO = avatarsList[i].avatarRenderer.lodRenderer.gameObject;
+                if (!lodGO.activeSelf)
+                    continue;
+
+                if (applyVerticalMovement)
+                    lodGO.transform.localPosition = new Vector3(lodGO.transform.localPosition.x, (lodGO.transform.localPosition.y > LODS_LOCAL_Y_POS ? LODS_LOCAL_Y_POS : LODS_LOCAL_Y_POS + LODS_VERTICAL_MOVEMENT), lodGO.transform.localPosition.z);
+
+                Vector3 previousForward = lodGO.transform.forward;
+                Vector3 lookAtDir = (lodGO.transform.position - CommonScriptableObjects.cameraPosition).normalized;
+
+                lookAtDir.y = previousForward.y;
+
+                lodGO.transform.forward = lookAtDir;
+            }
+
+            if (applyVerticalMovement)
+                lastLODsVerticalMovementTime = Time.timeSinceLevelLoad;
+        }
+
+        void OnMainPlayerReposition(Vector3 newPos, Vector3 previousPos) { UpdateAllLODs(); }
+
+        public void UpdateAllLODs()
```

Comment:
Is it possible to make this method private and call it when `lodDistance` or `maxNonLODAvatars` are changed? (converting them into properties)

Reply:
Done, now we use those config values from the DataStore and the setting control controllers set those values in there